### ISSUE TITLE
Mac OS and Windows installations do not appear

### ISSUE THREAD

**Issue by MaplesThroughWindows** [dakanji/RefindPlus#126]
### RefindPlus Version

v0.13.3.AB Release

### Device Type

Apple Mac

### Problem Description

I'm using MyBootMgr 086 to configure my cMP 5,1 but I'm running into _exactly_ the same problem as the OP in [issue #17](https://github.com/dakanji/RefindPlus/issues/17), right down to the screenshot.

In that thread you say:

> I see you have resolved the issue by selecting correct options in the tool provided in MyBootMgr.

_Originally posted by @dakanji in https://github.com/dakanji/RefindPlus/issues/17#issuecomment-765179274_

I can't find any reference to what the OP's configuration mistake was, however. I've answered all the questions in ConfigFactory correctly several times over and re-deployed/re-blessed the destination EFI with no change. I'm sure the mistake is probably mine somewhere, but I've no clue what it might be.

**Describe the bug**:
After following the instructions on the MacRumors thread and using MyBootMgr to configure my device with ConfigFactory, the graphical boot picker does not show either my native macOS Mojave install or the legacy Windows 10 install. I only see two OpenCore options (default and OC_VMM).

I am using a fast USB flash drive in ESP Mode for RefindPlus. The USB flash drive is inserted into one of the native USB 2 ports on the front of the cMP.  The USB flash drive is necessary because I don't have any other suitable volumes for testing RefindPlus on.

Choosing the default OpenCore leads to a boot picker that offers my macOS Mojave install, but choosing this option I never reach a login window, I can't explain why.  I see the Apple logo, then the screen goes blank and I have no output to my display.

I have tried MyBootMgr in Solo and Dual mode with the same result.

I tried manually activating "decline_esp_filter" in RefindPlus config.conf.  The result was the same.

I have tried refreshing the boot picker several times but nothing new is offered.

**To reproduce**
N/A

**Expected behaviour**
RefindPlus should offer to directly boot a legacy Windows 10 install and a macOS 10.14 install.

**Environment**
- MyBootMgr 086
- Hardware: cMP 5,1 (genuine, not a flashed 4,1); AMD 5xx GPU; single Xeon CPU; 18GB RAM; legacy Windows 10 on SSD in SATA Bay 1; macOS 10.14 Mojave on NVMe SSD via PCIe adapter card.
- OS: macOS Mojave 10.14.6

### Problem Point

Before starting a loader or tool

### Affected Items

All OS Loaders (First Row Items)

### Debug Log

[22v15q5153.log](https://github.com/dakanji/RefindPlus/files/10013742/22v15q5153.log)
[DebugLog_ConfigFactory.log](https://github.com/dakanji/RefindPlus/files/10013743/DebugLog_ConfigFactory.log)

**Comment by dakanji** [comment]
On Windows ...
Where do you have the BOOTCAMP instance installed? That is, which Bay/Port?

On Mac OS...
> Choosing the default OpenCore leads to a boot picker that offers my macOS Mojave install, but choosing this option I never reach a login window, I can't explain why. I see the Apple logo, then the screen goes blank and I have no output to my display.

Please share the OpenCore log file.

**Comment by dakanji** [comment]
RefindPlus is unable to find the legacy boot code on the Windows disk and unable to find the bootloader on the Mac OS disk. If OpenCore can, then RefindPlus should as well, but evidently not for some reason.

I do notice an oddity in that a WIndows `Basic Data Partition` was found in the Mac OS APFS Partition:
![image](https://user-images.githubusercontent.com/1128876/202581147-413adc7d-c86e-4ef8-9b23-d8b994fca2d4.png)
Unless an unlikely misindentification, that most certainly should not be there.

If you have spare disks, I would reinsall Mac OS and Windows ([Bare Metal](https://forums.macrumors.com/threads/2250317)), and see if things are different with those. BootCamp is a bit of a [nasty hack](https://www.rodsbooks.com/gdisk/hybrid.html)

Sorry, I can't offer more insight at this time

> Wanted to add that the OpenCore boot screen shows the Apple logo and a progress bar before the screen goes dark.

This is likely simply GPU related and that a different port is active after Mac OS is loaded. You can try switching to a different port.

> Quick question: when DeployConfig asks "Bless config on completion?" is it referring to a Soft Bless or a Hard Bless?

It is referring to whether it should automatically run BootBlesser

> Quick question: using BootBlesser to Soft Bless the config on the USB flash drive a second time and rebooting causes the boot process to hang. I have to repeat the DeployConfig process and then run BootBlesser to work around this. Is this expected behaviour?

Not particularly but also not clear what you exactly mean by `second time`. Can you set out the steps taken in bullet points?

**Comment by MaplesThroughWindows** [comment]
> This is likely simply GPU related and that a different port is active after Mac OS is loaded. You can try switching to a different port.

My monitor only supports DisplayPort and HDMI, and both of those produced no output after booting with OpenCore. If OpenCore is defaulting to my card's DVI port than I cannot verify this because my monitor has no DVI port. As I don't intend to use OpenCore initially I can ignore this until the more important issues are resolved and return to it later (if it even continues to be a problem at that stage).

> If you have spare disks, I would reinsall Mac OS and Windows ([Bare Metal](https://forums.macrumors.com/threads/2250317)), and see if things are different with those. BootCamp is a bit of a [nasty hack](https://www.rodsbooks.com/gdisk/hybrid.html)

I prepared a clean install of macOS 10.13.6 and tested MyBootMgr 086. The rest of the configuration remains the same, so all the same drives are present.

After running ConfigFactory, DeployConfig and BootBlesser from the clean macOS High Sierra installation, and using the same USB flash drive (though fully erased as GUID/HFS+ first) RefindPlus shows the macOS 10.13.6 install in the boot picker and I can boot from it.  The Windows and Mojave installations are still not presented.

Because you've pointed out the strange Basic Data Partition on the Mojave drive may be responsible for the problems there, I will prepare a clean Mojave installation later and test that with RefindPlus.

It's possible my legacy Windows installation was prepared through BootCamp so I will prepare a clean install of that too and test.

I don't know when I'll have time to prepare those clean installs, so if you would like to close this issue, I have opened a new one relating to my problems with the reliability of the Soft Bless procedure.
